src/video/videosurface.cpp

```diff
@@ -68,9 +70,14 @@ void VideoSurface::setSource(VideoSource *src)
     if (source == src)
         return;
 
-    unsubscribe();
-    source = src;
-    subscribe();
+    // To don't block GUI thread.
+    QtConcurrent::run([this, src]()
+    {
+        QMutexLocker l(&bigLock);

```

Comment:
I want to wrap `subscribe` in another thread because it call `openDevice()` for camera, which can load camera very long (and if had problem -- forever)